services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/map/ProcessedMapRepositoryImpl.java

```diff
@@ -0,0 +1,289 @@
+        MongoCollection<Document> collection = mongoTemplate.getCollection(collectionName);
+        DistinctIterable<Integer> docs = collection.distinct("properties.intersectionId", Integer.class);
+        List<IntersectionReferenceData> referenceDataList;
+        try (MongoCursor<Integer> results = docs.iterator()) {
+            referenceDataList = new ArrayList<>();
+            while (results.hasNext()) {
+                Integer intersectionId = results.next();
+                Bson projectionFields = Projections.fields(
+                        Projections.include("properties.intersectionId", "properties.originIp",
+                                "properties.refPoint.latitude", "properties.refPoint.longitude", "properties.intersectionName"),
+                        Projections.excludeId());
+                try {
+                    Document document = collection.find(eq("properties.intersectionId", intersectionId))
+                            .projection(projectionFields).sort(Sorts.descending("properties.timeStamp")).maxTime(props.getMongoTimeoutMs(), TimeUnit.MILLISECONDS).first();
+                    if (document != null) {
+                        IntersectionReferenceData data = new IntersectionReferenceData();
+                        Document properties = document.get("properties", Document.class);
+                        if (properties != null) {
+                            Document refPoint = properties.get("refPoint", Document.class);
+                            data.setIntersectionID(intersectionId);
+                            data.setRoadRegulatorID("-1");
+                            data.setRsuIP(properties.getString("originIp"));
+                            if (properties.getString("intersectionName") != null && properties.getString("intersectionName").isEmpty()) {
+                                data.setIntersectionName(properties.getString("intersectionName"));
+                            }
+                            if (refPoint != null) {
+                                data.setLatitude(refPoint.getDouble("latitude"));
+                                data.setLongitude(refPoint.getDouble("longitude"));
+                            }
+                        }
+                        referenceDataList.add(data);
+                    }
+                } catch (MongoException e) {
+                    logger.error("MongoDB Intersection Query Did not finish in allowed time window");
+                } catch (Exception e) {
+                    logger.error(e.getMessage());
+                }
+            }
+        }
+        return referenceDataList;
+    }
+
+    public List<IntersectionReferenceData> getIntersectionsContainingPoint(double longitude, double latitude){
+        MongoCollection<Document> collection = mongoTemplate.getCollection(collectionName);
+        DistinctIterable<Integer> docs = collection.distinct("properties.intersectionId", Integer.class);
```

Comment:
question: would it be more efficient to push the geospatial query onto Mongo rather than the multiple interactions with the database here?